python/cuml/preprocessing/model_selection.py

```diff
@@ -24,6 +24,109 @@
 from typing import Union
 
 
+def stratify_split(X, y, n_train, n_test, x_numba, y_numba, random_state):
+    """
+    Function to perform a stratified split based on y.
+    Identifies number of classes and samples per class, splices data within
+    each class.
+
+    Input:
+    X, y: shuffled input data and labels
+    """
+    if isinstance(X, cudf.DataFrame):
+        X = cp.fromDlpack(X.to_dlpack())
+    if isinstance(y, cudf.Series):
+        y = cp.fromDlpack(y.to_dlpack())
```

Comment:
Why is this using dlpack to convert to cupy as opposed to `X.values`? Furthermore, why is it converting dataframes (and Series) to cupy arrays if afterwards it has logic (using `append`) to process cudf objects? Wouldn't that logic go unused thanks to this conversion?

Reply:
Ah thanks for pointing this out. Although, I knew about `.values` (and forgot about it), the official docs don't have this listed https://docs.rapids.ai/api/cudf/nightly/10min-cudf-cupy.html - might make sense to update this?